### ISSUE TITLE
Proposal for language support

### ISSUE THREAD

**Issue by JonahKr** [rhasspy/rhasspy-hermes-app#1]
I created a component for homeassistant once and really likes the way they handle translations.
more detailed info >[here](https://developers.home-assistant.io/docs/translations)

Obviously we don't have the userbase to use platforms like Lokalize but this can just as easy be replaced by accepting pullrequests to your skill Repo.

However the intent Json contains no information about the language being used for each specific case. One ugly way would be reading it from the profil but we still wouldn't know which profile is the active one. Maybe it would be possible to add a "lang" field in the Intent JSON with the language abbreviation (like _en_ for english or _de_ for german).

**Comment by koenvervloesem** [comment]
Yes, I was thinking about something like this, but without the JSON (which is not very human-friendly).

I'm cosidering using Python's [gettext support](https://docs.python.org/3/library/gettext.html) for translating text before sending it to the TTS. This will also let non-developers contribute translations easier because many translation tools support gettext .po files.

For the sentences for an intent I would create a `sentences.ini` for the app specifically for each language.

This could result in a directory structure such like this:

```
lang
├── de
│   ├── app.po
│   └── sentences.ini
└── en
    ├── app.po
    └── sentences.ini
```

But for all this to happen transparently Rhasspy should indeed add a `lang` attribute to the relevant MQTT messages. I have opened an issue here: https://github.com/rhasspy/rhasspy-hermes/issues/11.

**Comment by JonahKr** [comment]
I didn't know about gettext...  But after reading through the docs a bit it definitely seems like a better(and already well implemented) option.

**Comment by maxbachmann** [comment]
I like this idea. Using something standardised like gettext as you proposed, since it is supported by most translation systems without any extra effort. So it would be possible to translate skill e.g. with weblate or similar web applications (essentially what mycroft does aswell).

For the sentences.ini file I think it would be good to be able to have multiple sentence and multiple slot files.

**Comment by EinfachArne** [comment]
Thanks a lot for the input regarding language support. I made a first implementation on my skill but the sentences.ini files are still missing because I am looking for a way to load them into Rhasspy on start.

https://github.com/EinfachArne/rhasspy-openhab-skill/tree/develop

**Comment by koenvervloesem** [comment]
@EinfachArne I actually wrote something quite similar a while ago while testing the new lang attribute in the Hermes messages, but your implementation is very nice! Maybe it's an idea to add your `getTranslator` function  to Rhasspy Hermes App.
Retraining the `sentences.ini` file on the fly with an MQTT message is indeed something I'd very much like (see https://github.com/rhasspy/rhasspy-hermes/issues/12). Then we can really make the process of installing new apps quite easy, with as less user interaction as possible.
